todo_app_e2e_tests/test_todo_e2e.py

```diff
@@ -0,0 +1,128 @@
+import os, pytest
+from threading import Thread
+from todo_app.data.todo import TodoService
+from todo_app import app
+from dotenv import find_dotenv, load_dotenv
+
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import TimeoutException
+
+
+
+@pytest.fixture(scope='module') 
+def app_with_temp_board():
+    # Create the new board & update the board id environment variable
+    os.environ['TRELLO_BOARD_NAME'] = 'E2E_Test_Board' 
+
+    todoService = TodoService()
+
+    board_id = todoService.board_id
+    print(f'board id = {board_id}')
+
+    # construct the new application
+    application = app.create_app() 
+
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True 
+    thread.start()
+    yield application
+
+    # Tear Down
+    thread.join(1) 
+    todoService.delete_board(board_id)
+
+@pytest.fixture(scope="module") 
+def driver():
+    # Load the app config
+    file_path = find_dotenv('.env')
+    load_dotenv(file_path, override=True)
+
+    #Set up headless option for chrome driver
+    chrome_options = webdriver.ChromeOptions()
+    chrome_options.add_argument('--headless')
+
+    with webdriver.Chrome(options=chrome_options) as driver:
+        yield driver
+
+@pytest.fixture(autouse=True) 
+def archive_cards():
+    os.environ['TRELLO_BOARD_NAME'] = 'E2E_Test_Board' 
+    todoService = TodoService()
+    todoService.archive_cards()
+
+def test_task_journey(driver, app_with_temp_board):
+    driver.get('http://localhost:5000')
+    assert driver.title == "To-Do App"
+
+def test_add_task_journey(driver, app_with_temp_board):
+    driver.get('http://localhost:5000')
+
+    task_title = driver.find_element_by_id('title')
+    task_title.send_keys("Selenium Test Add")
+    task_title.submit()
+    try:
+        todo_item_added = WebDriverWait(driver, 5).until(
+            EC.presence_of_element_located((By.ID, "todo_item_1")))
```

Comment:
You can actually just set up the driver with `driver.implicitly_wait(5)` and then commands like `driver.find_element_by_id` will wait up to that amount of time.

Reply:
The implicitly_wait() option is not working, tests fail if I use this option.
I used it like this
with webdriver.Chrome(options=chrome_options) as driver:
        driver.implicitly_wait(5)
        yield driver

Reply:
`todo_item_added = driver.find_element_by_id("todo_item_1")` seems to work fine for me. 

What causes the failure?

Reply:
at my end if I don't add the wait, my test fails to find the element.